desc/compute/_fast_ion.py

```diff
@@ -0,0 +1,355 @@
+        "B^phi",
+        "B^phi_r|v,p",
+        "|B|_r|v,p",
+        "b",
+        "grad(phi)",
+        "grad(psi)",
+        "|grad(psi)|",
+        "|grad(rho)|",
+        "|e_alpha|r,p|",
+        "kappa_g",
+        "iota_r",
+    ]
+    + Bounce2D.required_names,
+    resolution_requirement="tz",
+    grid_requirement={"can_fft2": True},
+    **_bounce_doc,
+)
+@partial(
+    jit,
+    static_argnames=[
+        "Y_B",
+        "num_transit",
+        "num_well",
+        "num_quad",
+        "num_pitch",
+        "pitch_batch_size",
+        "surf_batch_size",
+        "spline",
+    ],
+)
+def _Gamma_c(params, transforms, profiles, data, **kwargs):
+    """Fast ion confinement proxy as defined by Nemov et al.
+
+    Poloidal motion of trapped particle orbits in real-space coordinates.
+    V. V. Nemov, S. V. Kasilov, W. Kernbichler, G. O. Leitold.
+    Phys. Plasmas 1 May 2008; 15 (5): 052501.
+    https://doi.org/10.1063/1.2912456.
+    Equation 61.
+
+    A 3D stellarator magnetic field admits ripple wells that lead to enhanced
+    radial drift of trapped particles. The energetic particle confinement
+    metric γ_c quantifies whether the contours of the second adiabatic invariant
+    close on the flux surfaces. In the limit where the poloidal drift velocity
+    majorizes the radial drift velocity, the contours lie parallel to flux
+    surfaces. The optimization metric Γ_c averages γ_c² over the distribution
+    of trapped particles on each flux surface.
+
+    The radial electric field has a negligible effect, since fast particles
+    have high energy with collisionless orbits, so it is assumed to be zero.
+    """
+    # noqa: unused dependency
+    theta = kwargs["theta"]
+    Y_B = kwargs.get("Y_B", theta.shape[-1] * 2)
+    num_transit = kwargs.get("num_transit", 20)
+    num_pitch = kwargs.get("num_pitch", 64)
+    num_well = kwargs.get("num_well", Y_B * num_transit)
+    pitch_batch_size = kwargs.get("pitch_batch_size", None)
+    surf_batch_size = kwargs.get("surf_batch_size", 1)
+    assert (
+        surf_batch_size == 1 or pitch_batch_size is None
+    ), f"Expected pitch_batch_size to be None, got {pitch_batch_size}."
+    spline = kwargs.get("spline", True)
+    fieldline_quad = (
+        kwargs["fieldline_quad"] if "fieldline_quad" in kwargs else leggauss(Y_B // 2)
+    )
+    quad = (
+        kwargs["quad"]
+        if "quad" in kwargs
+        else get_quadrature(
+            leggauss(kwargs.get("num_quad", 32)),
+            (automorphism_sin, grad_automorphism_sin),
+        )
+    )
+
+    def Gamma_c(data):
+        bounce = Bounce2D(
+            grid,
+            data,
+            data["theta"],
+            Y_B,
+            num_transit,
+            quad=quad,
+            is_fourier=True,
+            spline=spline,
+        )
+
+        def fun(pitch_inv):
+            points = bounce.points(pitch_inv, num_well=num_well)
+            v_tau, drift1, drift2 = bounce.integrate(
```

Comment:
I want to know how difficult it would be to calculate these quantities as a function of alpha, pitch_inv if my grid contains multiple alpha values. I need to vectorize these calculations in alpha for Gamma_alpha, Gamma_delta.

Reply:
doing this now in different PR